### ISSUE TITLE
Incorrect probabilities when using with  custom trained yolov11m for VIDEO, works fine with Image

### ISSUE THREAD

**Issue by ambarishg** [ultralytics/ultralytics#20374]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Trained using yolov11m.pt

When inferring with images , the custom model works fine and the confidence scores for the bounding boxes are fine.

When using with a Video , the custom model gives incorrect confidence scores for some bounding boxes. This occurs only for some bounding boxes.

```
import torch
with torch.no_grad():
    results = model(source, stream=True, conf=0.35)

for result in results:
    img = result.orig_img  # Get the original image frame
    boxes = result.boxes    # Get bounding boxes
    probabilities = boxes.conf  # Get confidence scores  
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ambarishg, thank you for bringing this to our attention and for using Ultralytics 🚀! This is an automated response to help get you the support you need—an Ultralytics engineer will review your issue shortly.

We recommend checking our [Docs](https://docs.ultralytics.com/) for guidance on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. Many common questions are answered there.

Since this appears to be a 🐛 Bug Report, could you please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) if you haven't already? This will help us reproduce and diagnose the issue more efficiently. Include, if possible:
- The code snippet you use for both image and video inference
- A sample frame or short video where the problem occurs
- Model and environment details

If your question relates to training or data specifics, please share as much detail as possible and review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community for more help or to discuss with others:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) for community threads

## Upgrade

Please make sure you are using the latest version of the `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience and for helping us improve Ultralytics!

**Comment by ambarishg** [comment]
![Image](https://github.com/user-attachments/assets/2552e3eb-370c-4409-8fbf-cf14a39e3e80)

This is perfect for the Image.

![Image](https://github.com/user-attachments/assets/fd6fdaa2-e144-4632-b555-7712a8480fcf)

The same frame in the Video is giving the incorrect result

**Comment by Y-T-G** [comment]
Can you elaborate on your issue?

What you did and what the issue is?

**Comment by Y-T-G** [comment]
> I am using the SAME Custom Trained Model
> 
> ![Image](https://github.com/user-attachments/assets/2552e3eb-370c-4409-8fbf-cf14a39e3e80)
> 
> This is perfect for the Image.
> 
> ![Image](https://github.com/user-attachments/assets/fd6fdaa2-e144-4632-b555-7712a8480fcf)
> 
> The same frame in the Video is giving the incorrect result
> 

Can you show the prediction logs?

**Comment by Y-T-G** [comment]
Can you also show the output from Ultralytics from `result.plot()`?

**Comment by ambarishg** [comment]
FRAME 14
RESULTS ARE
ultralytics.engine.results.Results object with attributes:

boxes: ultralytics.engine.results.Boxes object
keypoints: None
masks: None
names: {0: 'BEES', 1: 'DRONES', 2: 'POLLENBEES'}
obb: None
orig_img: array([[[197, 197, 195],
        [199, 199, 197],
        [199, 199, 197],
        ...,
        [181, 175, 170],
        [181, 175, 170],
        [181, 175, 170]],

       [[197, 197, 195],
        [199, 199, 197],
        [199, 199, 197],
        ...,
        [181, 175, 170],
        [181, 175, 170],
        [181, 175, 170]],

       [[196, 197, 192],
        [196, 197, 192],
        [197, 198, 193],
        ...,
        [181, 175, 170],
        [181, 175, 170],
        [181, 175, 170]],

       ...,

       [[ 87,  89,  91],
        [ 83,  85,  87],
        [ 83,  85,  87],
        ...,
        [158, 195, 219],
        [158, 195, 219],
        [158, 195, 219]],

       [[ 90,  92,  94],
        [ 90,  92,  94],
        [ 90,  92,  94],
        ...,
        [158, 195, 219],
        [158, 195, 219],
        [158, 195, 219]],

       [[ 84,  86,  88],
        [ 85,  87,  89],
        [ 86,  88,  90],
        ...,
        [158, 195, 219],
        [158, 195, 219],
        [158, 195, 219]]], shape=(720, 1280, 3), dtype=uint8)
orig_shape: (720, 1280)
path: '/root/BEES02/bees-backend/uploads/Hive 89.mp4'
probs: None
save_dir: 'runs/detect/predict'
speed: {'preprocess': 1.1026859283447266, 'inference': 222.72086143493652, 'postprocess': 0.4570484161376953}
------------------------------

> PROB IS 0.9055247902870178
> CLASS IS BEES
> PROB IS 0.7326151132583618
> CLASS IS BEES

**Here this one is incorrect**

> PROB IS 0.7326151132583618
> CLASS IS BEES

**Comment by Y-T-G** [comment]
Did you train with JPG or PNG?

**Comment by ambarishg** [comment]
> Did you train with JPG or PNG?

JPG

**Comment by ambarishg** [comment]
Is this a BUG ?

**Comment by Y-T-G** [comment]
I don't think it's bug

**Comment by ambarishg** [comment]
Do we require any other information to solve this problem ? Please let me know of any details if I could help with

**Comment by Y-T-G** [comment]
What code did you use to get image prediction?

**Comment by ambarishg** [comment]
```
import torch
with torch.no_grad():
    results = model(source, stream=True, conf=0.35)

for result in results:
    img = result.orig_img  # Get the original image frame
    boxes = result.boxes    # Get bounding boxes
    probabilities = boxes.conf  # Get confidence scores  
```

The model is the Custom Trained Model.

For the image, it is working fine.

For the video , it is giving incorrect probabilities for some bounding boxes for the same frame.

**Comment by Y-T-G** [comment]
You used the same code for getting both image and video prediction?

**Comment by ambarishg** [comment]
Yes thats correct. I used the same code.

**Comment by Y-T-G** [comment]
How many images did you use for training?

**Comment by ambarishg** [comment]
400 images were used for training. 

I know this could have been more.

However, what I find surprising is that for the image the probability is working fine but for the video, since I had to use stream = true, the result is coming differently

**Comment by Y-T-G** [comment]
It just seems like overfitting due to small dataset.

**Comment by ambarishg** [comment]
Why would it work for an image and not for the same frame in the video ? Any pointers

**Comment by Y-T-G** [comment]
It means the model is affected by very small changes in appearance, such as compression artefacts which is a sign of overfitting.
